### ISSUE TITLE
Boot times on live media are too long

### ISSUE THREAD

**Issue by AJSlye** [manjaro/manjaro-tools#136]
As reported by Clemens, It currently takes over 45 seconds to boot in live mode.
These services seem to be the majority of the issue:
27.822s mhwd-live.service
9.754s livecd.service
3.993s pacman-init.service
3.824s dev-sdb1.device
2.244s ModemManager.service

**Comment by philmmanjaro** [comment]
Hi James,

**mhwd-live** is around 25s, which is normal. 9s for **livecd** is also normal. **pacman-init** checks all servers. Depending on the internet-speed it might take a while to get it done. Fastest bootup is always with free drivers and no internet connection. I'll take a look at it again.

**Comment by philmmanjaro** [comment]
This is a normal bootup as it gets with **Virtualbox**
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 8.127s (kernel) + 38.688s (userspace) = 46.816s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         14.300s mhwd-live.service
          9.309s livecd.service
          8.930s pacman-init.service
          3.744s plymouth-start.service
          2.375s ModemManager.service
          1.893s NetworkManager.service
          1.567s dev-sr0.device
           878ms polkit.service
           871ms systemd-vconsole-setup.service
           846ms bluetooth.service
           619ms avahi-daemon.service
           414ms systemd-modules-load.service
           294ms systemd-tmpfiles-setup-dev.service
           288ms systemd-journald.service
           275ms wpa_supplicant.service
           258ms systemd-journal-flush.service
           258ms alsa-restore.service
           246ms lightdm-plymouth.service
           243ms dev-hugepages.mount
           237ms systemd-udev-trigger.service
           231ms plymouth-quit-wait.service
           206ms systemd-sysctl.service
           191ms systemd-remount-fs.service
           183ms systemd-logind.service
           183ms udisks2.service
           165ms sys-kernel-debug.mount
           164ms systemd-udevd.service
           160ms dev-mqueue.mount
           154ms tmp.mount
           142ms plymouth-read-write.service
           137ms systemd-tmpfiles-setup.service
           111ms user@1000.service
           107ms kmod-static-nodes.service
            80ms systemd-hostnamed.service
            80ms ntpd.service
            75ms upower.service
            67ms systemd-update-utmp.service
            64ms accounts-daemon.service
            42ms rtkit-daemon.service
            38ms sys-kernel-config.mount
            29ms systemd-random-seed.service
            18ms systemd-user-sessions.service
            12ms etc-pacman.d-gnupg.mount
...skipping...
           231ms plymouth-quit-wait.service
           206ms systemd-sysctl.service
           191ms systemd-remount-fs.service
           183ms systemd-logind.service
           183ms udisks2.service
           165ms sys-kernel-debug.mount
           164ms systemd-udevd.service
           160ms dev-mqueue.mount
           154ms tmp.mount
           142ms plymouth-read-write.service
           137ms systemd-tmpfiles-setup.service
           111ms user@1000.service
           107ms kmod-static-nodes.service
            80ms systemd-hostnamed.service
            80ms ntpd.service
            75ms upower.service
            67ms systemd-update-utmp.service
            64ms accounts-daemon.service
            42ms rtkit-daemon.service
            38ms sys-kernel-config.mount
            29ms systemd-random-seed.service
            18ms systemd-user-sessions.service
            12ms etc-pacman.d-gnupg.mount
```

This is totally normal and expected. Never had any issues with it before.

**Comment by AJSlye** [comment]
Well the above is what I get on bare metal, Intel core 3 with 8 gigs of ram
and Intel only graphics with no wired internet connection, using the free
driver option.

**Comment by AJSlye** [comment]
Here is the entire output of systemd-analyze blame:

          27.822s mhwd-live.service
          9.754s livecd.service
          3.993s pacman-init.service
          3.824s dev-sdb1.device
          2.244s ModemManager.service
          1.426s NetworkManager.service
          1.174s polkit.service
          1.173s systemd-vconsole-setup.service
           735ms systemd-modules-load.service
           690ms plymouth-start.service
           616ms systemd-journald.service
           498ms avahi-daemon.service
           496ms systemd-udevd.service
           410ms systemd-tmpfiles-setup-dev.service
           348ms alsa-restore.service
           342ms systemd-udev-trigger.service
           271ms systemd-logind.service
           243ms systemd-binfmt.service
           218ms wpa_supplicant.service
           214ms plymouth-read-write.service
           202ms systemd-journal-flush.service
           201ms systemd-sysctl.service
           181ms sys-kernel-debug.mount
           178ms tmp.mount
           151ms bluetooth.service
           150ms dev-hugepages.mount
           147ms systemd-remount-fs.service
            92ms dev-mqueue.mount
            89ms udisks2.service
            83ms systemd-hostnamed.service
            83ms systemd-random-seed.service
            66ms systemd-tmpfiles-setup.service
            66ms kmod-static-nodes.service
            65ms rtkit-daemon.service
            52ms systemd-backlight@backlight:intel_backlight.service
            48ms systemd-update-utmp.service
            48ms user@1000.service
            26ms sys-kernel-config.mount
            24ms plymouth-quit.service
            24ms plymouth-quit-wait.service
            21ms systemd-user-sessions.service
            21ms etc-pacman.d-gnupg.mount
            20ms upower.service
            11ms sys-fs-fuse-connections.mount
            10ms proc-sys-fs-binfmt_misc.mount
             8ms systemd-rfkill@rfkill0.service
             2ms systemd-rfkill@rfkill1.service

**Comment by philmmanjaro** [comment]
Well, if I start the same image in EFI mode you have to wait around 1:28 min to have the XFCE desktop loaded. With BIOS it is around 1 minute. This is normal. EFI needs ca. 39 seconds to load the kernel image and display the first systemd output. This is a different story and was always present so far.

**Comment by AJSlye** [comment]
This is not an issue for me personally, but Clemens is used to faster boot times in Ubuntu.\
I don't want to see him to drop this project just because of boot times that are approx 20 sec longer than what he is used to for a live boot image. :)

**Comment by AJSlye** [comment]
This boot time issue started only since the last few updates to manjaro-tools, prior to that the media booted much, much faster.

**Comment by AJSlye** [comment]
However, the MHWD service was much shorter prior to the fix for the dynamic linker cache issue:

**Comment by philmmanjaro** [comment]
Well, Ubuntu has all drivers pre-installed. So no package manager needs to be triggered to verify the package and install it. Additionally the configuration time is not there. To speed it up you have to create additional overlays you load with having the drivers pre-installed on them. This results in more overlay and bigger install medias as mostly the given drivers collide with each other. In the past I tried that to speed up the boot-process but was dismissed by the community. I'll check now an older install media to compare the speed against **Manjaro 15.12-alpha2** I already posted above.

**Comment by AJSlye** [comment]
Please do.
Personally, I think if the times are comparable maybe it's just more
noticeable now that it only shows these few services loading and doesn't
scroll the list of what is happening in between like it used to. It could
also be that Clemens may be seeing the number on the mhwd-live service
counter and assuming that it is only this service taking that amount of
time, and not realizing that it's the last service to load and starts at 20
something sec. and ends at 40 something.

**Comment by philmmanjaro** [comment]
I just booted up **Manjaro 0.8.0** which took around one minute to load the XFCE desktop. However **systemd-analyze** is there broken due missing packages. I'll try another one from 2013.

**Comment by AJSlye** [comment]
I wonder why your getting 14 sec and on my system i'm getting 27 with the same ISO?
Could it be due to actual hardware verses virtual-box?

**Comment by philmmanjaro** [comment]
I really see no issues in speed changes over time:
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 4.230s (kernel) + 38.838s (userspace) = 43.068s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         19.626s livecd.service
         15.265s mhwd.service
          2.999s NetworkManager.service
          2.502s plymouth-start.service
          1.087s systemd-vconsole-setup.service
           914ms systemd-modules-load.service
           725ms polkit.service
           685ms systemd-logind.service
           631ms systemd-tmpfiles-setup-dev.service
           556ms systemd-udev-trigger.service
           507ms sys-kernel-debug.mount
           466ms dev-hugepages.mount
           464ms tmp.mount
           464ms dev-mqueue.mount
           460ms systemd-remount-fs.service
           317ms udisks2.service
           310ms rtkit-daemon.service
           241ms systemd-sysctl.service
           228ms plymouth-read-write.service
           204ms haveged.service
           197ms upower.service
           110ms alsa-restore.service
           108ms systemd-tmpfiles-setup.service
           101ms systemd-update-utmp.service
            93ms systemd-udevd.service
            89ms bluetooth.service
            88ms sys-kernel-config.mount
            81ms systemd-random-seed-load.service
            76ms sys-fs-fuse-connections.mount
            53ms plymouth-quit.service
            50ms plymouth-quit-wait.service
            47ms systemd-journal-flush.service
            31ms systemd-user-sessions.service
[manjaro@manjaro Desktop]$ inxi -F
System:    Host: manjaro Kernel: 3.10.11-1-MANJARO i686 (32 bit) Desktop: Xfce 4.10.2 Distro: ManjaroLinux 0.8.7.1 Ascella
```

However this image has no plymouth enabled ...

**Comment by AJSlye** [comment]
Question, How is it that  livecd.service is longer than mhwd.service on that ISO and the opposite on the newer ones? 
Anyway, I see that mhwd.service has always taken around 15+ seconds in virtualbox.
I'm just wondering what do you get booting the ISO on the hardware itself?

**Comment by philmmanjaro** [comment]
We have one speciality with **Manjaro 0.8.4**:
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 4358ms (kernel) + 31432ms (userspace) = 35790ms
[manjaro@manjaro Desktop]$ systemd-analyze blame
  4277ms NetworkManager.service
  1469ms systemd-logind.service
  1134ms systemd-vconsole-setup.service
   951ms systemd-modules-load.service
   486ms tmp.mount
   472ms dev-hugepages.mount
   457ms udisks2.service
   436ms dev-mqueue.mount
   407ms sys-kernel-debug.mount
   392ms systemd-udev-trigger.service
   363ms polkit.service
   335ms systemd-udevd.service
   276ms systemd-tmpfiles-setup.service
   276ms systemd-sysctl.service
   273ms systemd-remount-fs.service
   155ms modem-manager.service
   153ms upower.service
   123ms bluetooth.service
    87ms sys-fs-fuse-connections.mount
    80ms rtkit-daemon.service
    64ms sys-kernel-config.mount
    52ms rc-local.service
    19ms systemd-user-sessions.service
     9ms proc-sys-fs-binfmt_misc.mount
[manjaro@manjaro Desktop]$ inxi -S
System:    Host: manjaro Kernel: 3.7.9-1-MANJARO i686 (32 bit) Desktop: Xfce 4.10.0 Distro: Manjaro Linux
```
![systemd-boot-0 8 4](https://cloud.githubusercontent.com/assets/5050022/9977400/09526f94-5ef6-11e5-9449-023ca1cdc849.jpg)

Over time the technique changed. You can see that in the plot how long each service took. Systemd got also more and more complex over time. This delays it also. Sometimes the hardware installation was inside of **livecd.service** on newer ones we splitted it up. Also changing from **aufs** to **overlayfs** might have changed some stuff. However the complete bootup regarding **mhwd** mostly didn't changed.

**Comment by Kirek** [comment]
mhwd.service at a point was broken (not all xorg packages needed by mhwd where in the livecd), so mhwd did nothing because there weren't all the stuff there, and the only reason it booted at all is because the xorg server ships with the modesetting driver by default... but yes it was faster.

**Comment by Kirek** [comment]
We could probably slash a second or two in the livecd.service if we generate all locales when building the iso.

**Comment by philmmanjaro** [comment]
Also just a notice: **i686** boots faster than **x86_64** as those also load **lib32 drivers**.

Here is the current take on **0.8.12 i686**:

```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 7.864s (kernel) + 31.950s (userspace) = 39.814s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         14.075s ldconfig.service
         10.721s livecd.service
          9.228s pacman-init.service
          5.965s mhwd.service
          1.494s NetworkManager.service
          1.476s ModemManager.service
          1.352s systemd-udev-hwdb-update.service
          1.059s plymouth-start.service
           920ms systemd-vconsole-setup.service
           707ms systemd-modules-load.service
           641ms polkit.service
           523ms systemd-binfmt.service
           517ms systemd-logind.service
           511ms rpcbind.service
           491ms systemd-journald.service
           464ms systemd-sysctl.service
           383ms tmp.mount
           382ms systemd-journal-flush.service
           382ms dev-mqueue.mount
           375ms dev-hugepages.mount
           356ms plymouth-read-write.service
           334ms systemd-journal-catalog-update.service
           310ms systemd-remount-fs.service
           308ms systemd-sysusers.service
           307ms systemd-udev-trigger.service
           296ms sys-kernel-debug.mount
           243ms bluetooth.service
           234ms lightdm-plymouth.service
           205ms plymouth-quit-wait.service
           154ms systemd-tmpfiles-setup-dev.service
           132ms upower.service
           130ms alsa-restore.service
           120ms systemd-tmpfiles-setup.service
           119ms systemd-udevd.service
           106ms etc-pacman.d-gnupg.mount
           105ms pacman-boot.service
           101ms pamac.service
            90ms udisks2.service
            83ms proc-sys-fs-binfmt_misc.mount
            72ms user@1000.service
            65ms kmod-static-nodes.service
            50ms accounts-daemon.service
            41ms systemd-update-utmp.service
            39ms systemd-random-seed.service
            36ms sys-kernel-config.mount
            26ms rtkit-daemon.service
            19ms systemd-user-sessions.service
            18ms systemd-update-done.service
             4ms sys-fs-fuse-connections.mount
...skipping...
           243ms bluetooth.service
           234ms lightdm-plymouth.service
           205ms plymouth-quit-wait.service
           154ms systemd-tmpfiles-setup-dev.service
           132ms upower.service
           130ms alsa-restore.service
           120ms systemd-tmpfiles-setup.service
           119ms systemd-udevd.service
           106ms etc-pacman.d-gnupg.mount
           105ms pacman-boot.service
           101ms pamac.service
            90ms udisks2.service
            83ms proc-sys-fs-binfmt_misc.mount
            72ms user@1000.service
            65ms kmod-static-nodes.service
            50ms accounts-daemon.service
            41ms systemd-update-utmp.service
            39ms systemd-random-seed.service
            36ms sys-kernel-config.mount
            26ms rtkit-daemon.service
            19ms systemd-user-sessions.service
            18ms systemd-update-done.service
             4ms sys-fs-fuse-connections.mount
[manjaro@manjaro Desktop]$ inxi -S
System:    Host: manjaro Kernel: 3.16.7.4-1-MANJARO i686 (32 bit)
           Desktop: Xfce 4.11.8git-UNKNOWN Distro: ManjaroLinux 0.8.12 Ascella
```
![systemd-analyze-0 8 12](https://cloud.githubusercontent.com/assets/5050022/9977465/a157e614-5ef8-11e5-9e58-d2e61db05a99.jpg)

So on my end it is just fine as always. Why does it take much longer on hardware?
Simple: a CD/DVD is slower in booting up. On USB-Sticks it depends on the transfer speed. SD cards might be faster. And with VirtualBox the ISO will be loaded directly from RAM. Hope this clears it up.

And yes, in the past we didn't generate any locales at all. This was an issue in the past. Generating all locales will result in bigger install medias and make no sense, at least for me.

**Comment by udeved** [comment]
Its no problem with openrc.
livecd takes about 30 secs to boot into DE.

**Comment by AJSlye** [comment]
Clemens just posted this video to an issue tracker for netrunner rolling: https://github.com/netrunner-rolling/netrunner-rolling-issues/issues/1

He wan't the ISO to be error free.

**Comment by philmmanjaro** [comment]
Ramon added some changes to make more services run parallel. However, did you tried **openrc** yet. Marcus told me already it is sometimes faster than **systemd**. Most of the posted issues are not present with this init system.

**Comment by AJSlye** [comment]
I'll rebuild Manjaro-tools-git to incorporate the changes.
I guess I can ask if Clemens would like me to try openrc, but I'm not sure how that will go.
Please take a look at the issue tracker link I posted so you can see what he's asking me to do.

**Comment by Kirek** [comment]
The changes are in the manjaro-tools-livecd package:
https://github.com/manjaro/manjaro-tools-livecd/commit/f2f495335c2503fbc69f9eb90e2ff35e13ccf856
Grab it from unstable or build it yourself.

**Comment by philmmanjaro** [comment]
Well, we didn't change **manjaro-tools** but **manjaro-livecd**. You can find those packages in our **unstable** branch. I'll keep track on it.

**Comment by Kirek** [comment]
mhwd-live will still take a long time thought, but at least it will start a little earlier

**Comment by philmmanjaro** [comment]
You might also disable **plymouth** so Clemens sees what is going on. When he is fine with it you can put it back on ;)

**Comment by AJSlye** [comment]
I used the manjaro-tools-git pkgbuild to build the devel branch so it builds all three packages at once. Please take a look at the link I posted.

**Comment by Kirek** [comment]
"switch_root: failed to unlink ro_branch: Directory not empy" message has been always in Manjaro, no idea how to supress or fix it, but its never the cause for a halted boot (but because of quiet boot it may be the last message you see).

**Comment by Kirek** [comment]
the manjaro-tools-livecd is a different PKGBUILD...

**Comment by AJSlye** [comment]
I already tried disabling plymouth and removing quite from grub, it still only shows the three services.

**Comment by AJSlye** [comment]
Oh OK, I was thinking of iso not livecd. I'll build an ISO using unstable to see the difference then.

**Comment by AJSlye** [comment]
If you watch the video he is hitting esc to exit plymouth anyway, that's the other issue he want's me to fix,
"plymouthd: could not start boot splash: no such file or directory", I can't fix that one either and it happens on the Manjaro KDE ISO as well. I'm pretty sure this has something to do with plymouthd and systemd, but I'm not positive what causes this as it works on shutdown.

**Comment by philmmanjaro** [comment]
Latest **Manjaro 15.12** has following times:
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 6.899s (kernel) + 25.996s (userspace) = 32.895s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         20.463s mhwd-live.service
         12.810s livecd.service
          9.124s pacman-init.service
          3.860s plymouth-start.service
          2.579s ModemManager.service
          1.753s NetworkManager.service
          1.454s dev-sr0.device
          1.072s polkit.service
           684ms systemd-vconsole-setup.service
           499ms wpa_supplicant.service
           381ms alsa-restore.service
           368ms avahi-daemon.service
           327ms systemd-modules-load.service
           320ms systemd-journald.service
           290ms systemd-udev-trigger.service
           289ms systemd-tmpfiles-setup-dev.service
           252ms systemd-logind.service
           214ms lightdm-plymouth.service
           198ms plymouth-quit-wait.service
           188ms systemd-journal-flush.service
           183ms systemd-sysctl.service
           174ms udisks2.service
           172ms dev-hugepages.mount
           169ms tmp.mount
           157ms sys-kernel-debug.mount
           147ms systemd-udevd.service
           142ms ntpd.service
           134ms dev-mqueue.mount
           111ms systemd-tmpfiles-setup.service
           104ms plymouth-read-write.service
           102ms systemd-remount-fs.service
            97ms pamac.service
            81ms user@1000.service
            74ms systemd-random-seed.service
            66ms upower.service
            63ms accounts-daemon.service
            60ms rtkit-daemon.service
            50ms sys-kernel-config.mount
            32ms kmod-static-nodes.service
            24ms systemd-update-utmp.service
            24ms etc-pacman.d-gnupg.mount
            16ms systemd-user-sessions.service
...skipping...
           188ms systemd-journal-flush.service
           183ms systemd-sysctl.service
           174ms udisks2.service
           172ms dev-hugepages.mount
           169ms tmp.mount
           157ms sys-kernel-debug.mount
           147ms systemd-udevd.service
           142ms ntpd.service
           134ms dev-mqueue.mount
           111ms systemd-tmpfiles-setup.service
           104ms plymouth-read-write.service
           102ms systemd-remount-fs.service
            97ms pamac.service
            81ms user@1000.service
            74ms systemd-random-seed.service
            66ms upower.service
            63ms accounts-daemon.service
            60ms rtkit-daemon.service
            50ms sys-kernel-config.mount
            32ms kmod-static-nodes.service
            24ms systemd-update-utmp.service
            24ms etc-pacman.d-gnupg.mount
            16ms systemd-user-sessions.service
[manjaro@manjaro Desktop]$ inxi -S
System:    Host: manjaro Kernel: 4.1.7-1-MANJARO x86_64 (64 bit)
           Desktop: Xfce 4.12.3 Distro: ManjaroLinux 15.12-dev Capella
```

![mj-1512-optimized](https://cloud.githubusercontent.com/assets/5050022/9978803/d97ba076-5f39-11e5-8ce7-4191f927646a.jpg)

Well, it is faster than the last build.

**Comment by philmmanjaro** [comment]
So here are the latest **Capella** boots in both architectures using **VirtualBox**. I think faster we don't get anymore with the current **development** state:
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 6.735s (kernel) + 25.719s (userspace) = 32.455s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         20.303s mhwd-live.service
         12.386s livecd.service
          8.887s pacman-init.service
          3.704s plymouth-start.service
          2.177s ModemManager.service
          1.294s NetworkManager.service
          1.203s dev-sr0.device
          1.107s polkit.service
           683ms systemd-vconsole-setup.service
           600ms wpa_supplicant.service
           478ms systemd-modules-load.service
           343ms systemd-udev-trigger.service
           338ms avahi-daemon.service
           309ms systemd-journald.service
           243ms systemd-tmpfiles-setup-dev.service
           220ms udisks2.service
           217ms systemd-journal-flush.service
           203ms systemd-logind.service
           192ms lightdm-plymouth.service
           174ms plymouth-quit-wait.service
           168ms systemd-udevd.service
           143ms systemd-sysctl.service
           133ms plymouth-read-write.service
           117ms alsa-restore.service
           112ms kmod-static-nodes.service
           108ms tmp.mount
           108ms ntpd.service
            98ms systemd-tmpfiles-setup.service
            85ms user@1000.service
            84ms dev-hugepages.mount
            80ms dev-mqueue.mount
            79ms systemd-random-seed.service
            74ms systemd-remount-fs.service
            63ms sys-kernel-debug.mount
            60ms upower.service
            58ms rtkit-daemon.service
            56ms etc-pacman.d-gnupg.mount
            54ms accounts-daemon.service
            47ms systemd-update-utmp.service
            15ms systemd-user-sessions.service
            12ms sys-kernel-config.mount
            10ms sys-fs-fuse-connections.mount
...skipping...
           174ms plymouth-quit-wait.service
           168ms systemd-udevd.service
           143ms systemd-sysctl.service
           133ms plymouth-read-write.service
           117ms alsa-restore.service
           112ms kmod-static-nodes.service
           108ms tmp.mount
           108ms ntpd.service
            98ms systemd-tmpfiles-setup.service
            85ms user@1000.service
            84ms dev-hugepages.mount
            80ms dev-mqueue.mount
            79ms systemd-random-seed.service
            74ms systemd-remount-fs.service
            63ms sys-kernel-debug.mount
            60ms upower.service
            58ms rtkit-daemon.service
            56ms etc-pacman.d-gnupg.mount
            54ms accounts-daemon.service
            47ms systemd-update-utmp.service
            15ms systemd-user-sessions.service
            12ms sys-kernel-config.mount
            10ms sys-fs-fuse-connections.mount
[manjaro@manjaro Desktop]$ inxi -S
System:    Host: manjaro Kernel: 4.1.7-1-MANJARO x86_64 (64 bit)
           Desktop: Xfce 4.12.3 Distro: ManjaroLinux 15.12-dev Capella

[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 7.665s (kernel) + 26.361s (userspace) = 34.027s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         20.337s mhwd-live.service
         13.891s pacman-init.service
         12.939s livecd.service
          3.765s plymouth-start.service
          2.296s ModemManager.service
          2.008s NetworkManager.service
          1.381s polkit.service
          1.282s dev-sr0.device
           664ms systemd-logind.service
           662ms wpa_supplicant.service
           660ms systemd-vconsole-setup.service
           525ms bluetooth.service
           477ms systemd-journald.service
           341ms systemd-modules-load.service
           341ms systemd-tmpfiles-setup-dev.service
           221ms avahi-daemon.service
           215ms systemd-udev-trigger.service
           211ms systemd-sysctl.service
           209ms lightdm-plymouth.service
           197ms plymouth-read-write.service
           197ms plymouth-quit-wait.service
           188ms alsa-restore.service
           164ms systemd-udevd.service
           162ms ntpd.service
           156ms systemd-journal-flush.service
           123ms udisks2.service
           113ms systemd-hostnamed.service
            94ms systemd-random-seed.service
            86ms user@1000.service
            82ms upower.service
            80ms systemd-tmpfiles-setup.service
            72ms tmp.mount
            71ms accounts-daemon.service
            68ms pamac.service
            66ms sys-kernel-debug.mount
            64ms kmod-static-nodes.service
            60ms dev-mqueue.mount
            48ms systemd-update-utmp.service
            43ms systemd-remount-fs.service
            43ms sys-kernel-config.mount
            39ms dev-hugepages.mount
            33ms rtkit-daemon.service
            18ms etc-pacman.d-gnupg.mount
            17ms systemd-user-sessions.service
            10ms sys-fs-fuse-connections.mount
...skipping...
           164ms systemd-udevd.service
           162ms ntpd.service
           156ms systemd-journal-flush.service
           123ms udisks2.service
           113ms systemd-hostnamed.service
            94ms systemd-random-seed.service
            86ms user@1000.service
            82ms upower.service
            80ms systemd-tmpfiles-setup.service
            72ms tmp.mount
            71ms accounts-daemon.service
            68ms pamac.service
            66ms sys-kernel-debug.mount
            64ms kmod-static-nodes.service
            60ms dev-mqueue.mount
            48ms systemd-update-utmp.service
            43ms systemd-remount-fs.service
            43ms sys-kernel-config.mount
            39ms dev-hugepages.mount
            33ms rtkit-daemon.service
            18ms etc-pacman.d-gnupg.mount
            17ms systemd-user-sessions.service
            10ms sys-fs-fuse-connections.mount
[manjaro@manjaro Desktop]$ inxi -S
System:    Host: manjaro Kernel: 4.1.7-1-MANJARO i686 (32 bit)
           Desktop: Xfce 4.12.3 Distro: ManjaroLinux 15.12-dev Capella
```

**Comment by Kirek** [comment]
I spend some time checking how much time it takes to run every item in the livecd script:
```
Loading scripts - Time: 0.02 minutes
Get consolefont and arch - Time: 0.00 minutes
Updated fstab for swap - Time: 0.02 minutes
Language configured - Time: 0.00 minutes
Run locale-gen - Time: 0.03 minutes
Configure env - Time: 0.00 minutes
Deleting existing machine-id ...
Setting machine-id ...
Configure machine-id - Time: 0.00 minutes
Configure sudoers - Time: 0.00 minutes
Configured extramodules - Time: 0.13 minutes
Configured displaymanager autologin - Time: 0.00 minutes
Configured root user - Time: 0.00 minutes
Configured alsa - Time: 0.00 minutes
```
Seems the big item is this line:
depmod $(cat /usr/lib/modules/$(uname -r)/extramodules/version)
What exactly it does?

**Comment by philmmanjaro** [comment]
Hmm, it was introduced with **[0.8.2](https://github.com/manjaro/manjaroiso/commit/aac492b23c34e431723ce749152ebf34eea949f2)** back in **2012**. I've to check if this is still needed.

**Comment by philmmanjaro** [comment]
@Kirek: can you bring your time code into the livecd.log. Would be nice to have it displayed like this.

**Comment by philmmanjaro** [comment]
Even with enhanced tuning the total boot-up time varies always around 30 - 35s:
```
[manjaro@manjaro Desktop]$ systemd-analyze
Startup finished in 6.851s (kernel) + 26.837s (userspace) = 33.689s
[manjaro@manjaro Desktop]$ systemd-analyze blame
         21.591s mhwd-live.service
          8.925s pacman-init.service
          7.953s livecd.service
          3.657s plymouth-start.service
          2.441s ModemManager.service
          1.677s NetworkManager.service
          1.268s dev-sr0.device
          1.065s polkit.service
```

**Comment by Kirek** [comment]
I doubt we can reduce this much more unless we change how we configure the graphics drivers.

**Comment by udeved** [comment]
> can you bring your time code into the livecd.log. Would be nice to have it displayed like this.

Please make sure that this code is only run with systemd.

**Comment by philmmanjaro** [comment]
Why, this code is universal.

**Comment by udeved** [comment]
How? systemd-analyse is not present on openrc system.

**Comment by philmmanjaro** [comment]
We don't use **systemd-analyze** for our log files. We use **date**. I just posted my results by using **systemd-analyze** to have a proper result.

**Comment by udeved** [comment]
Is the systmd boot time ok now?

**Comment by philmmanjaro** [comment]
See the posted stats here. Yes, it is faster now.

**Comment by udeved** [comment]
So can we close it?

**Comment by philmmanjaro** [comment]
Let AJSlye decide that. From our end it is what we could do so far. Everything else would be a redesign of how **mhwd** works.

**Comment by AJSlye** [comment]
I wouldn't be the one to decide this, Clemens would.
I do agree with Philip here, anything else would require me to find a way around using MHWD on the ISO and load the all the xorg drivers by default, which would probably be more work than it's worth.

**Comment by udeved** [comment]
I could readd a feature that can install x11 stuff in a chroot instead of iso repo and on the fly.
So there would be a x11drivers image instead of mhwd repo.

**Comment by AJSlye** [comment]
That would not only be cool but a big source of my headaches gone.
If this could be enabled or disabled (managed) by a true/false switch in the manjaro-tools.conf file as well that would be awsome.

**Comment by udeved** [comment]
Right, thats the idea, a switch.
I will do this in x11 branch until merged with devel branch.

**Comment by Kirek** [comment]
You can try that now, you only have to remove stuff from Packages-Mhwd and put them in Packages-live and see how it goes.

**Comment by udeved** [comment]
The question is, do we want an optional x11 drivers layer instead of mhwd repo?
Mr AjJ needs to say.

**Comment by Kirek** [comment]
No, the question is: will it work ? Not only boot, and not only for graphic drivers, but also vm, networks drivers and other stuff. And calamares/thus also runs a mhwd script that depends on this mhwd repo.
It's not that simple.

**Comment by udeved** [comment]
Is there mhwd on netrunner? I never tested the netrunner.

**Comment by Kirek** [comment]
Of course is it, its based on manjaro...

**Comment by AJSlye** [comment]
This is ultimately your distro and tools, all I can do is give you an excerpt from an email sent to me from Clemens:
"It takes 38 seconds on my intel gfx machine with 4gb ram and recent cpu.
Even 20 seconds are intolerable, so lets get rid of this and revert back to whatever any other distro but manjaro does."

If it would work then having this as an option would be useful. As far as calameres and thus, I wouldn't suggest changing the way they work. The issue is currently only on the live media, not the installed system.

**Comment by AJSlye** [comment]
The Rolling Edition of Netrunner is currently based on Manjaro, Netrunner standard edition is based on Kubuntu. The issue is that the kubuntu based live media is truculently booting faster than the Manjaro based ISO. This shouldn't be the case, Manjaro should be lighter and faster right?

**Comment by udeved** [comment]
I guess the mhwd repo on iso approach is simply a bit slow as it is.
Its not much that can be done there from manjaro-tools, except a mentioned optional layer.
But, I think this would need some changes in mhwd and how it configures drivers.

**Comment by Kirek** [comment]
As we said the problem is how mhwd does it thing, the main problem even if we preinstall all the packages-mhwd stuff, mhwd still insist on reinstalling them, so we don't gain anything unless we make mhwd a little smarter.

**Comment by Kirek** [comment]
@AJSlye 
This problem is in the live media, in the installed system doesn't run mhwd-live, that's why we basically ignored this issue for a long time.

**Comment by udeved** [comment]
I noticed that the time of mhwd live script increases, if I enable prop drivers, ie the repo gets bigger, mor packages.
Only free drivers is faster.

**Comment by AJSlye** [comment]
Right, how many chipsets does MHWD really need to look for?
 I'm sure we only need to look for Intel, Nvidia, ATI/AND and possibly sis and VIA, I mean how many people actually have neomagic, trident, tseng, etc. any longer?

**Comment by Kirek** [comment]
Looking for the correct drivers takes no time, what it happen is that mhwd runs pacman to install the correct packages and that is what it takes time (unpacking them...)

**Comment by philmmanjaro** [comment]
Well, **mhwd** is already smart and will skip the packages. We also have the feature already included into **manjaroiso** from the get go. I simply disabled it due the fact that the ISO is bigger. However, there is no need to edit **mhwd** at all. We simply agreed to "drop" that feature for the always released **I install only the needed stuff**.

**Comment by udeved** [comment]
If I understand right, if we reintroduce an optional chroot drivers image, the iso would boot considerably faster, but the iso size increases naturally.
I would be of use for AJ.

**Comment by AJSlye** [comment]
Yea, the ISO size doesn't really mater since the older ISO's Leszek made by extracting your release ISO and modifying them we're already 2.4 gig. The ISO's I've been making using Manjaro-tools are only 1.9 gig, so there is room for expansion.

**Comment by Kirek** [comment]
I'm testing now what happens if we preinstall all the packages that mhwd needs.

**Comment by udeved** [comment]
@AJSlye 

I will make that happen in a drvopt branch.
Just modify later the git build to use the brach, while kirek tests the mhwd behaviour

**Comment by AJSlye** [comment]
OK, sounds good. Will there be an option I need to modify in manjaro-tools.conf or profile.conf?

**Comment by philmmanjaro** [comment]
Here we go for the history:

* [disable xorg-overlays on all editions](https://github.com/manjaro/manjaroiso/commit/57b781783b7ae8732d1d981d3a800a59668f6f40)
* [split free and non-free drivers](https://github.com/manjaro/manjaroiso/commit/5d6a187cfc83f133021ddb857e1639f0f08a2c95#diff-e8615a4cebc83c08ba92839709e4be1c)
* [further splitting of xorg-packages](https://github.com/manjaro/manjaroiso/commit/ff4644155365f3ec8579e4f14fb49443209ea9ad)

I think **0.8.1** was the only release we did so far using that feature. So how does it work?

* you need one overlay for free drivers
* another one for proprietray drivers
* adjust **isolinux** and any other livecd bootloader to adopt those changes with the kernel cmd **overlay=**

You might have now conflicts as we didn't had Catalyst back then. For free drivers it is easy to be done. Remember, some packages are already preinstalled like **ndiswrapper**.

The feature to have ultra fast bootup was always there, but since **2012** disabled. I never felt the need to have it as we blew **CD-size** always. Back then we wanted to have the best **live-cd** with a most packages but all drivers on the disk not pre-installing all driver. And this goal has to be kept. This is the only feature which makes **Manjaro** **Manjaro**. Not installing unneeded stuff from the get-go. However for **live-session** it doesn't matter. It simply will skip the installations.

I also recommend to pre-install only **nvidia** and let the other packages go. Clemens only needs the free drivers preinstalled to have his **ultra fast** bootup.

Why I kept it from you? Well, nobody read the given code or asked me. I also never thought this will ever pop up again as we always did it the way we did it.

**Comment by philmmanjaro** [comment]
@AJSlye: at the moment it is disabled and not used for years. I've to dig up my old code and re-introduce that feature. It won't make it for **15.09** release. I can try but I can't promise. This was always ultra low for me and I was always against it to do so. I simply don't give a shit what others think about it. Simply like that. It always bothered me how the other distros had to uninstall things. This was designed for **Chakra** and adopted by **Manjaro** and I won't change it. I only might re-introduce the feature to have **some preinstalled drivers** for the live-session.

**Comment by AJSlye** [comment]
Yes, I think for the live-session only as well, the installed system wouldn't need this.

**Comment by udeved** [comment]
@AJSlye 

I implemented a mhwd_repo switch.
It will be in the profile.conf

**Comment by Kirek** [comment]
We don't really need to make major changes, it's a matter to add the drivers we want in the Package-Live list (or merge two list) this way we don't need to make extra overlay images.

**Comment by philmmanjaro** [comment]
No, we need additional overlays, otherwise we have all the "shit" preinstalled. That is not the way to go. You can add it for ease to the **Livecd-Overlay** package list. I recommend to add:

* all free graphical drivers
* all free network drivers
* nvidia as prorietary

Never add them to your **Desktop-Package-List**.

**Comment by udeved** [comment]
On mhwd_repo=false list merge, mhwd with livecd, and skip mhwd repo?
I think an additiona layer is no problem, perhaps more flexible also

**Comment by philmmanjaro** [comment]
@udeved: no, we need **mhwd** overlay always. You can only add additional drivers to the Livecd-Overlay package list. Thats it. See the post above.

**Comment by Kirek** [comment]
This is the results, in my test it was usually between 30 and 40s before.
[manjaro@manjaro ~]$ systemd-analyze
Startup finished in 7.168s (firmware) + 158ms (loader) + 7.456s (kernel) + 7.543s (userspace) = 22.327s
[manjaro@manjaro ~]$ systemd-analyze blame                                                            
          4.709s pacman-init.service                                                                  
          4.008s mhwd-live.service                                                                    
          3.514s livecd.service                                                                           
          2.158s dev-sr0.device                                                                           
          1.498s plymouth-start.service                                                                   
          1.412s ModemManager.service                                                                         
          1.085s NetworkManager.service                                                                       
           760ms systemd-logind.service                                                                       
           742ms systemd-modules-load.service                                                                    
           741ms avahi-daemon.service                                                                            
           728ms systemd-vconsole-setup.service                                                                   
           578ms bluetooth.service                                                                                
           457ms polkit.service                                                                                     
           389ms wpa_supplicant.service                                                                              
           304ms systemd-journald.service                                                                             
           249ms systemd-udev-trigger.service                                                                           
           209ms systemd-journal-flush.service                                                                            
           206ms systemd-udevd.service                                                                                     
           198ms upower.service
           196ms systemd-tmpfiles-setup-dev.service
           143ms udisks2.service
           126ms systemd-remount-fs.service
           125ms tmp.mount
           122ms dev-hugepages.mount
           120ms alsa-restore.service
           119ms plymouth-read-write.service
           112ms systemd-tmpfiles-setup.service
           109ms systemd-sysctl.service
           101ms dev-mqueue.mount
            99ms systemd-hostnamed.service
            79ms sys-kernel-debug.mount
            43ms rtkit-daemon.service
            40ms user@1000.service
            34ms systemd-update-utmp.service
            24ms plymouth-quit.service
            19ms plymouth-quit-wait.service
            14ms sys-kernel-config.mount
            13ms systemd-random-seed.service
            13ms kmod-static-nodes.service
             9ms systemd-user-sessions.service
             5ms etc-pacman.d-gnupg.mount

**Comment by philmmanjaro** [comment]
Why do we need the **mhwd-overlay**? Simple: we need it for **offline installations** as not always internet is present. If the driver is not present **mhwd** will try to download it or fails. Pre-installed drivers will simply skip the installation step in **mhwd**, it stills configures the drivers, though.

**Comment by udeved** [comment]
I was thinking we simply have the switch, and if AJ wants it to be preinstalled, he simply deactivates mhwd_repo switch, and all the drivers in Packages-Mhwd will be installed. Almost no changes, just to install instead of making repo.

**Comment by AJSlye** [comment]
Yes, we need just pre-install the drivers on the live session, but don't remove the overlay, calamares and thus need these if an internet connection is not available.

**Comment by AJSlye** [comment]
Oops, sorry, I hit the wrong button.

**Comment by udeved** [comment]
Ok, then its basically no changes and you fiddle with your Livecd list then. :)

**Comment by philmmanjaro** [comment]
See, mhwd is already fast. It was always the package manager ;)
```
         21.591s mhwd-live.service
          8.925s pacman-init.service
          7.953s livecd.service
```

```
          4.709s pacman-init.service                                                        
          4.008s mhwd-live.service                                                          
          3.514s livecd.service
```

@udeved: No, you still don't get it. **mwhd-overlay** is needed for offline installs. All drivers provided by **mhwd-db** needs to be present in that overlay. There is no switch needed at all. **manjaro-tools** has the feature already. Simply add those **pre-installed** drivers to your **Livecd-Overlay-Package** list. Thats it.

**Comment by udeved** [comment]
See last post.

**Comment by AJSlye** [comment]
Yes, I think I get it. Ramon is that all you did when you tested to get 4 sec. instead of 20+?

**Comment by philmmanjaro** [comment]
Yep, your to fast for me ;). Hope everybody got it now and we can close this one here. @udeved. I'll add a new present file so everybody gets it. @AJSlye simply link that then as your **live-cd-package** list.

**Comment by Kirek** [comment]
@AJSlye 
Just copy everything from Packages-Mhwd to Packages-Live
You can put the whole list of Packages-Mhwd but only ONE nvidia nonfree driver (you will need to remove 304 and 320.

**Comment by AJSlye** [comment]
OK, thanks philip.
I think if we can get 4 - 5 sec instead of 20+ Clemens will be satisfied. ;)

**Comment by AJSlye** [comment]
I think I'd rather leave the proprietary drivers even nvidia for booting with that option from syslinux.
I think I'll only need the free drivers installed on the ISO.

**Comment by Kirek** [comment]
I'm thinking that maybe locale-gen can be optimized. Afaik we generate en/de locales always in the iso, so if the user select this languages at boot (or in efi its always en_US), we could skip that:
```
Loaded scripts - Time: 0.062ms
Got consolefont and arch - Time: 0.037ms
Activated swap and added to fstab - Time: 0.291ms
Languaged configured - Time: 0.117ms
Ran locale-gen - Time: 1.716ms
Configured env - Time: 0.012ms
Deleting existing machine-id ...
Setting machine-id ...
Configured machine-id - Time: 0.055ms
Configured sudoers - Time: 0.028ms
Configured displaymanager autologin - Time: 1441508013.720ms
Configured root user - Time: 0.310ms
Configured alsa - Time: 0.231ms
```

**Comment by philmmanjaro** [comment]
Most boot it anyway for German or English. So this won't change anything. We can still go the road and do more locales. It always results in a bigger install media. We might add a config file for that, so everybody can decide if he wants to go that road or not.

**Comment by Kirek** [comment]
This locales are already generated in the images, so if the user select this languages we can skip regenerating them. It will only be for those two.

**Comment by philmmanjaro** [comment]
Ah ok, go for it.

**Comment by philmmanjaro** [comment]
And this also shows how the status was back then in **2012** and now, couple years later. Keep in mind to make it always optional and fully flexible so the maintainer can decide if he wants it. Also we should start to document all those features. Else you have always to ask me or read all the given code. For me it will be also harder as now more people contribute to our tools. Hope this will also show how important it is to document. And never try to drive me nuts. I really hate to explain my past decisions why I felt them that way.

**Comment by philmmanjaro** [comment]
Only two open issues for [0.9.13](https://github.com/manjaro/manjaro-tools/milestones/0.9.13). We are getting close now.

**Comment by udeved** [comment]
Why not achieve the same but have besides mhwd repo image a drivers image optional using same mhwd list?

**Comment by philmmanjaro** [comment]
Why do we need another overlay? **livecd-overlay** is exactly the overlay for packages we have just for the **live-session**. Those drivers are also only needed for that session? So adding another overlay will make no sense. However, if we want to have a switch, then yes - another overlay would make sense. However each maintainer should decide if he wants to have a small ISO or a bigger one but fast boot-up. Keep it kiss. Why should we add something which is obviously not needed anymore.

**Comment by AJSlye** [comment]
I agree with philip, if we can speed this up without making things overly complex then that is the rout to take.

**Comment by philmmanjaro** [comment]
I had this discussion years ago. And it was almost the same. So to solve this: we simply provide a solution both sides will get happy. On my end: I really don't need it. However we might use it for the **official supported** editions. LxQt and other maintainers won't use it at all.

**Comment by udeved** [comment]
Its clean in terms of package lists. 
The thought was, why not, we use something called overlayfs.
It should give same size. I don't want to force it, but I have a branch ready so we can simply test.

**Comment by AJSlye** [comment]
Yes, but does this branch deal with the fact that Calamares and Thus need the mhwd-overlay to install properly when offline?

**Comment by udeved** [comment]
Yes, we would just have an additional layer on switch, which is just the mhwd packages also installed besides the repo layer.
The only difference is additional layer, but no changes to livecd lists.

**Comment by philmmanjaro** [comment]
We can test your ideas for sure. However we should decide it as a team. I see your approach on this one. Re-inventing the wheel all the time over and over again is not the way to go. This is now the fast solution. If we get a smarter one, then fine. But please let this endless discussion go. I get more and more tired regarding this matter. I always designed my tools to be flexible for almost all cases.

The facts are:

* mhwd-overlay can't go (for sake of offline installs)
* those packages should never go to a desktop package list
* they have to have a valid overlay
* they are only used to speed up the boot process
* this is totally optional

**Comment by philmmanjaro** [comment]
@udeved: please check my provided file as it differs from the **mhwd** package list for given reasons. Else I'm fine with it. If it makes you happy, then go for it - but at least get somebody test it before merging it into **devel**. And please ask the team before you change some stuff to your liking. I always see some dudes get angry cos of that fact. We are a team and no one-man show anymore.

**Comment by AJSlye** [comment]
OK Phillip, I'm cloning manjaro-tools-profiles now to merge your changes to my profiles.

**Comment by udeved** [comment]
~~~
	if [[ -f Packages-Mhwd ]] ; then
		load_pkgs 'Packages-Mhwd'
		make_image_mhwd
		${mhwd_inst} && make_image_mhwd_drv
	fi
~~~

This is the idea. Totally optional.
Listen guys, I want to provide the best solution, and not the fastest, it may not be my one, its why I said, I have it in separate branch. I am personally totally neutral on this one, because I will stay with standard for my profile.

**Comment by philmmanjaro** [comment]
@udeved: the package list is not the same. We need another one. Look before you post something. The idea is ok, but you should start listen to other also. And read those posts again. With that current draft you will ran into package conflicts.

And here is another bug report which will do the other thing. Extend the boot-time: https://github.com/manjaro/manjaro-tools/issues/140

So all in all this will and will be always an never ending story ...

**Comment by AJSlye** [comment]
Philip, would I just remove Package-Livecd and replace it with packages-Livecd-Xorg?

**Comment by udeved** [comment]
I just don't understand the exclusive either or, I mean you can change livecd list, but some people may want to have preinstalled drivers for whatever reason in a separate image.  Making the drivers image loading a new list is not the issue I think.

**Comment by philmmanjaro** [comment]
@AJSlye A soft-link to that file would be also ok. If you don't have the shared folder, then yes - replace the provided one with yours.

**Comment by AJSlye** [comment]
Will, the softlink be followed, the name of the link changes with the file. I keep both shared and Netrunner in there own separate folder than what I download from git since I changed syslinux as well.

**Comment by philmmanjaro** [comment]
I've no exclusive either or topic here. I provided a fast and working solution to this matter. You can use that provided file and get rid of the normal **live-session** packages and call it **Packages-MHWD-Live** or what ever. You can't add all **proprietary drivers** to it. So I only added **nvidia** as this one is the most used driver. All the other drivers will be still installed **old-school**. If it is now in **livecd-overlay** or in another one - it would be the same. 

However you have to modify **isomounts** and so on. So more modifications are needed to get it work the way you like it. If you want to have each proprietary driver fast-boot, several overlays are needed and a smarter boot-screen with more options. All in all it will get more complex as simply adding those drivers to **Packages-Livecd**.

I'm open for all but I'm slowly but soon get sick about this situation talking about it again and again to come to no point. If you have the time to go the extra mile, then do it. I don't hinder you. But please include the team into it as everybody else does too.

**Comment by philmmanjaro** [comment]
@AJSlye now it feels more like Kindergarden. Did you checked your profile folder? There are several files and folders symlinked to **shared**. On my end I've only **Packages-Xfce** and the **profile.conf** as real files. Either I'm to long on a PC or you all drive me nuts some how ...

**Comment by udeved** [comment]
Where is the formular to file to start a branch?

**Comment by AJSlye** [comment]
Yes when I changed the link from Packages-Livecd to Pakages-livecd-xorg the link name will change as well. I was just wondering if that would make a difference as far as buildiso was concerned. I mean if the link changes would manjaro-tools still follow it? I'm just concerned that Manjaro tools wont recognize Pakages-livecd-xorg and skip it even if it's a softlink, that's all.

**Comment by philmmanjaro** [comment]
@udeved for real? Do I really need to make some Popcorn now?

```
git checkout -b <whatever>
```
This is common sense to create a new branch if you want to try out something. Even in **Calamares** we do it.

**Comment by udeved** [comment]
You mistake a little joke.
Calm down please, and we can settle whatever it is per email.

**Comment by philmmanjaro** [comment]
I'm calm. I even don't started yet. Seems it was a long day for me. Good news is, the current git-checkout created my **dev-build** just fine. Will install it now to see how it went.

**Comment by AJSlye** [comment]
Since either way should work and I'm keeping my profiles seperate from the rest I'll just replace livecd with livecd-xorg. Thanks phillip, i'm going to test this out tonight. ;D +1:

**Comment by philmmanjaro** [comment]
@AJSlye hope this will make Clemens happy. What ever the outcome will be, we have found a proper solution for the boot-time matter by now. As it is a hot topic it might even pop up again. Try to boot Ubuntu from CD. It takes for ages ;)

**Comment by AJSlye** [comment]
Yes, I know. And how is it that kubuntu so much faster than Ubuntu at boot times, etc. this makes no sense to me. I thought unity and their upstart/systemd hybrid solution was supposed to be so much better than this. LOL

**Comment by philmmanjaro** [comment]
Magic ;) - Well I can't find any regressions so far (https://github.com/manjaro/manjaro-tools/issues/141). We are good to go. Will merge now the current packages with **testing**. **manjaro-tools** should be still used from git. I still have to decide how we provide a snapshot of all compatible profiles for that package. Else the community will complain again that the master profiles don't work. Maybe reintroducing the **devel** branch might help. However our maintainers should commit then to the right branch. This is and will be always tricky.

**Comment by Kirek** [comment]
I don't think we made any profile breaking change in this cycle.

And going back to boot times, right now the slower service is `pacman-init`.
Is there any reason we initialize the pacman keys at boot instead of doing it in buildiso?

**Comment by AJSlye** [comment]
Man, I just started building against testing. LOL.

**Comment by philmmanjaro** [comment]
@Kirek I've to check why I introduced **pacman-init**. It should be also possible to do it within creation of the **root-image**. If I remember it correctly we had a bug and needed to create it during bootup. It is always good to optimize it. Mostly I left a note with the commit in **pacman** ...

**Comment by philmmanjaro** [comment]
@AJSlye you simply worry to much. Create the image and tell Clemens the dudes of Manjaro might even reduce the boot-speed even more. However you will have already the half time needed as before ...

**Comment by AJSlye** [comment]
I was mostly making a joke,  but yea, I know.

**Comment by philmmanjaro** [comment]
@Kirek simple - security reasons: initializes a new random GnuPG key on every boot.
So in the past we had the issue that every **Manjaro install media** had the same key. So everybody had the same key. So we introduced it to create one on bootup and copy that key to the to-be-installed system.

We can do now this:

* use a common used key for the live-session (as it really don't matter which key is used there) and create a new one with our installers
* or leave it as it is right now

I'd say, fuck the 4 - 8 seconds we gain out of it for security reasons.

**Comment by Kirek** [comment]
Ok that makes sense, we will keep as it is.

**Comment by udeved** [comment]
8 secs on systemd to init pacman?
is there other service started inbetween haveged and pacman-init?

**Comment by AJSlye** [comment]
Yes, that was an approx number as systemd initializes services in parallel.

**Comment by udeved** [comment]
Doesn't take that long on openrc with parallel boot enabled.

**Comment by philmmanjaro** [comment]
Well, **haveged** needs to be started first then it gens a random number and then **pacman-key** kicks in. If you don't use **haveged** it takes for ages. You even have to move your mouse or type buttons to get a random number generated. This also points out that **openrc** is in some ways the better **init-system** :P. @AJSlye: recommend it to Clemens and see what he thinks about it ... Does the power features fully work on KDE? @udeved ...

**Comment by AJSlye** [comment]
I actually haven't got around to asking that one yet. I wan't to do a test build with openrc on my end first.

**Comment by AJSlye** [comment]
OK, the resulting ISO is only 1990 Meg, this is not so bad actually.

**Comment by philmmanjaro** [comment]
Great. Now you have to create a plot of your boot-time and post both, the one before and after to see the difference ;) Then we can go all to bed ...

**Comment by udeved** [comment]
I just thought, perhaps the unit files can be somehow optimized, if the parallel funnily slows it down.
If...

**Comment by philmmanjaro** [comment]
**systemd** is sometimes a **blackbox** ;) Welcome to **MacOS**

**Comment by AJSlye** [comment]
OK, not sure about plotting it but now get:

systemd-analyze blame
          5.990s mhwd-live.service
          5.985s livecd.service
          5.929s pacman-init.service
          4.345s dev-sdb.device
          2.275s ModemManager.service
          1.957s nmbd.service
          1.774s NetworkManager.service
          1.199s polkit.service
          1.133s systemd-modules-load.service
          1.127s plymouth-start.service
          1.056s systemd-vconsole-setup.service
           868ms zramswap.service
           680ms smbd.service
           562ms systemd-journald.service
           436ms alsa-restore.service
           436ms systemd-tmpfiles-setup-dev.service
           386ms systemd-udevd.service
           334ms avahi-daemon.service
           320ms systemd-udev-trigger.service
           319ms systemd-sysctl.service
           316ms systemd-logind.service
           285ms wpa_supplicant.service
           255ms bluetooth.service
           231ms systemd-journal-flush.service
           218ms udisks2.service
           209ms sys-kernel-debug.mount
           204ms tmp.mount
           203ms dev-hugepages.mount
           195ms systemd-binfmt.service
           192ms upower.service
           176ms plymouth-read-write.service
           174ms systemd-remount-fs.service
           158ms kmod-static-nodes.service
           146ms systemd-hostnamed.service
           129ms dev-mqueue.mount
           120ms systemd-tmpfiles-setup.service
            96ms packagekit.service
            69ms systemd-random-seed.service
            68ms systemd-update-utmp.service
            57ms systemd-backlight@backlight:intel_backlight.service
            55ms user@1000.service
            47ms proc-sys-fs-binfmt_misc.mount
            38ms rtkit-daemon.service
            23ms sys-kernel-config.mount
            13ms etc-pacman.d-gnupg.mount
            12ms plymouth-quit.service
            12ms systemd-user-sessions.service
            11ms plymouth-quit-wait.service
             9ms systemd-rfkill@rfkill0.service
             7ms sys-fs-fuse-connections.mount
             5ms systemd-rfkill@rfkill1.service

             
total:
systemd-analyze
Startup finished in 15.431s (kernel) + 9.895s (userspace) = 25.326s

Before is posted previously: https://github.com/manjaro/manjaro-tools/issues/136#issuecomment-141684239

**Comment by AJSlye** [comment]
The strange thing is, now nmbd and smdb are actually loading when they we're not before, even though I had them added to my profile

**Comment by philmmanjaro** [comment]
Is that a good thing? To plot you would do `systemd-analyze plot > output.svg` and create a jpg, png out of it ;)

**Comment by Kirek** [comment]
systemd-analyze blame is not a full list of loaded services...

Panellizing pacman-init, livecd and mhwd-live is one of the first optimizations I did after this issue was opened and reduced the boot times for @philmmanjaro from 42-46s  to 32s... so there is no issue there.

**Comment by AJSlye** [comment]
OK, I didn't know that one. I'll do that when I get back then.

**Comment by AJSlye** [comment]
Thanks for the explanation there Kirek, that makes sense now. ;)

**Comment by philmmanjaro** [comment]
Yes, blame only lists those services which take the longest. However with plot you see which service was started when and how long it actually took in which context ...

**Comment by AJSlye** [comment]
Philip, do I really need the nvidia drivers in the new packages-livecd-xorg?
I'd rather leave these detected and load the old way, or would that cause an issue?

**Comment by Kirek** [comment]
No, you don't need them.

**Comment by AJSlye** [comment]
@Kirek
I had removed the question, but thanks for answering it anyway. That is good to know.

**Comment by AJSlye** [comment]
OK, I removed the nvidia driver from being installed to the live-session as well as thus and the cli installer and now the ISO is back to fitting on a 2 gig usb stick with 7 megs to spare (1920 MB).

**Comment by philmmanjaro** [comment]
@AJSlye did you got any feedback from Clemens? I assume we can close this one here now ...

**Comment by AJSlye** [comment]
I'm uploading a 2015.09-RC1 ISO based on these changes now, I will get feed
back possibly tomorrow.

**Comment by philmmanjaro** [comment]
Fine. no rush there ;)

**Comment by AJSlye** [comment]
If you would like to test the ISO I just pushed to the Blue Systems repository, then please do so, but as always do not share this to the public unless Clemens give the go ahead.

**Comment by philmmanjaro** [comment]
**Netrunner 15.09-rc1 (2015-09-24)**

Here is my current analyzis. Somehow Netrunner is slower in bootup than Manjaro KDE-Minimal 15.12-dev

```
[netrunner@netrunner ~]$ systemd-analyze
Startup finished in 8.679s (kernel) + 16.472s (userspace) = 25.151s
[netrunner@netrunner ~]$ systemd-analyze blame
         12.062s mhwd-live.service
         11.708s livecd.service
         11.315s pacman-init.service
          5.101s ModemManager.service
          4.162s nmbd.service
          4.131s NetworkManager.service
          3.568s dev-sr0.device
          2.289s polkit.service
          2.150s zramswap.service
          1.916s systemd-vconsole-setup.service
          1.595s systemd-modules-load.service
          1.458s smbd.service
          1.289s avahi-daemon.service
          1.049s alsa-restore.service
          1.011s bluetooth.service
           761ms systemd-sysctl.service
           525ms systemd-journald.service
           518ms systemd-udevd.service                                                             
           508ms wpa_supplicant.service                                                            
           503ms systemd-logind.service                                                            
           483ms systemd-tmpfiles-setup-dev.service                                                
           456ms packagekit.service                                                                
           444ms plymouth-start.service                                                            
           424ms systemd-binfmt.service                                                            
           340ms systemd-udev-trigger.service                                                      
           293ms systemd-journal-flush.service                                                     
           275ms sys-kernel-debug.mount                                                            
           275ms dev-mqueue.mount                                                                  
           271ms tmp.mount                                                                         
           270ms systemd-remount-fs.service                                                         
           268ms dev-hugepages.mount
           265ms ntpd.service
           249ms systemd-tmpfiles-setup.service
           224ms udisks2.service
           196ms systemd-hostnamed.service
           174ms kmod-static-nodes.service
           168ms plymouth-read-write.service
           167ms upower.service
            85ms systemd-update-utmp.service
            84ms proc-sys-fs-binfmt_misc.mount
            78ms user@1000.service
            78ms rtkit-daemon.service
            44ms sys-kernel-config.mount
            43ms plymouth-quit-wait.service
            41ms plymouth-quit.service
            30ms systemd-random-seed.service
            20ms etc-pacman.d-gnupg.mount
            19ms sys-fs-fuse-connections.mount
            17ms systemd-user-sessions.service
...skipping...
           525ms systemd-journald.service
           518ms systemd-udevd.service
           508ms wpa_supplicant.service
           503ms systemd-logind.service
           483ms systemd-tmpfiles-setup-dev.service
           456ms packagekit.service
           444ms plymouth-start.service
           424ms systemd-binfmt.service
           340ms systemd-udev-trigger.service
           293ms systemd-journal-flush.service
           275ms sys-kernel-debug.mount
           275ms dev-mqueue.mount
           271ms tmp.mount
           270ms systemd-remount-fs.service
           268ms dev-hugepages.mount
           265ms ntpd.service
           249ms systemd-tmpfiles-setup.service
           224ms udisks2.service
           196ms systemd-hostnamed.service
           174ms kmod-static-nodes.service
           168ms plymouth-read-write.service
           167ms upower.service
            85ms systemd-update-utmp.service
            84ms proc-sys-fs-binfmt_misc.mount
            78ms user@1000.service
            78ms rtkit-daemon.service
            44ms sys-kernel-config.mount
            43ms plymouth-quit-wait.service
            41ms plymouth-quit.service
            30ms systemd-random-seed.service
            20ms etc-pacman.d-gnupg.mount
            19ms sys-fs-fuse-connections.mount
            17ms systemd-user-sessions.service
[netrunner@netrunner ~]$ 
```

![netrunner-15 09-2015-09-24](https://cloud.githubusercontent.com/assets/5050022/10085389/7bb79f68-62f8-11e5-93ee-158a4a6db915.jpg)

**Comment by AJSlye** [comment]
Yes, Netrunner Rolling is far from minimal.
We added smbd and nmbd to the start-systemd= line because Clemens wanted samba-mounter working out of the box. And of course zramswap as well or it wouldn't be Netrunner.

**Comment by philmmanjaro** [comment]
Here is the analysis from **Manjaro 15.12-dev (2015-09-24)**:

```
[manjaro@manjaro ~]$ systemd-analyze
Startup finished in 8.055s (kernel) + 10.941s (userspace) = 18.997s
[manjaro@manjaro ~]$ systemd-analyze blame
          7.990s pacman-init.service
          6.000s livecd.service
          5.511s mhwd-live.service
          2.033s ModemManager.service
          1.681s NetworkManager.service
          1.419s dev-sr0.device
           876ms bluetooth.service
           856ms systemd-vconsole-setup.service
           757ms systemd-modules-load.service
           571ms polkit.service
           548ms plymouth-start.service
           490ms wpa_supplicant.service
           390ms avahi-daemon.service
           386ms systemd-journald.service
           354ms systemd-udevd.service
           261ms systemd-logind.service
           243ms systemd-tmpfiles-setup-dev.service
           209ms systemd-udev-trigger.service
           176ms systemd-journal-flush.service
           169ms tmp.mount
           166ms dev-hugepages.mount
           164ms alsa-restore.service
           164ms udisks2.service
           153ms dev-mqueue.mount
           138ms systemd-remount-fs.service
           135ms sys-kernel-debug.mount
           123ms systemd-tmpfiles-setup.service
           111ms plymouth-read-write.service
           103ms upower.service
            95ms user@1000.service
            86ms systemd-sysctl.service
            83ms systemd-hostnamed.service
            50ms systemd-update-utmp.service
            39ms rtkit-daemon.service
            36ms plymouth-quit-wait.service
            30ms systemd-random-seed.service
            30ms plymouth-quit.service
            28ms etc-pacman.d-gnupg.mount
            24ms sys-kernel-config.mount
            18ms systemd-user-sessions.service
             4ms kmod-static-nodes.service
...skipping...
           856ms systemd-vconsole-setup.service
           757ms systemd-modules-load.service
           571ms polkit.service
           548ms plymouth-start.service
           490ms wpa_supplicant.service
           390ms avahi-daemon.service
           386ms systemd-journald.service
           354ms systemd-udevd.service
           261ms systemd-logind.service
           243ms systemd-tmpfiles-setup-dev.service
           209ms systemd-udev-trigger.service
           176ms systemd-journal-flush.service
           169ms tmp.mount
           166ms dev-hugepages.mount
           164ms alsa-restore.service
           164ms udisks2.service
           153ms dev-mqueue.mount
           138ms systemd-remount-fs.service
           135ms sys-kernel-debug.mount
           123ms systemd-tmpfiles-setup.service
           111ms plymouth-read-write.service
           103ms upower.service
            95ms user@1000.service
            86ms systemd-sysctl.service
            83ms systemd-hostnamed.service
            50ms systemd-update-utmp.service
            39ms rtkit-daemon.service
            36ms plymouth-quit-wait.service
            30ms systemd-random-seed.service
            30ms plymouth-quit.service
            28ms etc-pacman.d-gnupg.mount
            24ms sys-kernel-config.mount
            18ms systemd-user-sessions.service
             4ms kmod-static-nodes.service
```
![manjaro-15 12-dev-2015-09-24](https://cloud.githubusercontent.com/assets/5050022/10085694/1c3ca6d0-62fa-11e5-9111-076b247f1f36.jpg)

**Comment by AJSlye** [comment]
We also have some deviations for the ksystemsettings5, plasma shell and default panel that slows down the Plasma desktop from loading a bit. Netrunner makes quite a few default settings and modifications.

**Comment by AJSlye** [comment]
Yes, but it's only a 7 sec difference and Clemens just emailed me that the 25 sec. it's taking now is acceptable compared to the 45-50 sec he was previously getting.

**Comment by AJSlye** [comment]
Oh yea, we also added sys to the addgroups+ line so that we don't need a root password for cups administration.

**Comment by AJSlye** [comment]
If there are no further Issues I'm marking this closed.

**Comment by udeved** [comment]
Just for info, an iso with preinstalled drivers layer and repo layer with only free drivers increases iso +10MB. 585MB vs 595MB lxqt-minimal

**Comment by philmmanjaro** [comment]
Free drivers are always small. The big part are those proprietary ones ...

**Comment by udeved** [comment]
I know, but its rather attractive for free drivers only withonly 10MB increase.
Boots fast and nice.

**Comment by AJSlye** [comment]
udeved, why are you changing manjaro-tools?
We can not use your switch unless it keeps the mhwd layer since calamares requires the mhwd repository for offline installs. Also, this seems like a waste of your time and resources since we already have a working fix in place.

**Comment by udeved** [comment]
Huh?
It is 
a) a separate branch
b) nothing changes except you get an additional layer with preinstalled drivers 
c) it has the same effect as adding the drivers to Packages-Livecd, no switches, only triggered if Packages-Mhwd-drivers is present

So whats the problem?

**Comment by AJSlye** [comment]
not really a problem but why wast time on something that doesn't need another layer to accomplish?

**Comment by udeved** [comment]
Because its much easier to test out this stuff if it is in a separate file and layer, instead of adding and removing drivers from livercd list for each build.

**Comment by AJSlye** [comment]
You don't, you just need to either softlink to Packages-Livecd or Packages-Livecd-xorg no need to add and remove anything from the lists. The only other change I nmade for my ISO was to remarked out the nvidia proprietary packages in Packages-Livecd-Xorg

**Comment by AJSlye** [comment]
I'm not trying to argue with you. If Philip and the rest of the maintainers think what your changing is OK then so do I.

**Comment by udeved** [comment]
Yeah, you could drop your xorg specific livecd list this way.
I mean, whats the deal, absolutely nothing changes. 
There is more than one method to achieve the end result you want.

**Comment by AJSlye** [comment]
Right and if the way we are doing it now works then why waste effort unless the new way somehow is more optimized.

**Comment by udeved** [comment]
Why not? If its wasted is kind of subjective.
Nobody pays me, I do it for fun doing it, imagine that.
I could build your profile this way, and would have netrunnner additions in a separate layer.

**Comment by philmmanjaro** [comment]
Well, remember that we need a different set of packages since proprietrary drivers will conflict with each other. This means, if there are the same packages as with **mhwd** repository image then it won't work. As of now I didn't had time to check your branch yet. I'm open for the change, but it should work and make somehow some sense way we put extra time into it, as the current set already works.

**Comment by udeved** [comment]
There aren't, its Packages-Mhwd-drivers.

**Comment by AJSlye** [comment]
Does this keep the mhwd-livecd repository layer required for offline installs or doesn't it?
Does this only install the free drivers to a separate layer?
Installing proprietary drivers creates an issue if older cards need legacy drivers as now it needs to remove drivers to install drivers or conflicts will occur, so more time added to boot ISO  when choosing non-free option. I had already closed this topic since it was fixed, do I need to re-open it now?

**Comment by udeved** [comment]
Yes, it does keep the repo, I said it already twice.
It installs whatever you put in the Packages-Mhwd-drivers.
Why would you reopen? Its in a branch, that is not about to be merged the next days I think.

**Comment by AJSlye** [comment]
Yes but yuo also made changes to the main branch of manjaro-tools-profiles as well, this too should have been in a separate branch I think.

**Comment by AJSlye** [comment]
Also why not install these drivers as well as add the repository in the same mhwd-live layer or in the livecd layer like we do now only adding a switch to turn it on or off instead of adding yet another layer?
Why make it more complex than it needs to be?

**Comment by udeved** [comment]
Its only a matter of defining Packages-Mhwd-drivers, no separate branch needed for profiles.

I was told using the same mhwd package list is not good.
Its trivial to change the install location, but I don't see any disadvantage using additional layers.
My test iso doesn't boot slower with additional layer.

**Comment by AJSlye** [comment]
OK, so how many layers or overlay file-systems are we up to already? How many is too many? What is the norm?

**Comment by udeved** [comment]
I personally don't care for any norm, and afaik there is no norm with overlayfs on iso.

We use rootfs and desktopfs, which amounts to one layer that is installed.
All the rest are live layers.

**Comment by AJSlye** [comment]
I'm just wondering, I'm not against this BTW, I just want to be sure this is not just wasted effort. What philip provided does the job well enough for me.

**Comment by udeved** [comment]
As said, I don't see it as wasted, its my time. :)

I am evaluating if I will use an additional layer for m lxqt-minimal profile.
mhwd-scripst runs fast with preinstalled drivers.

**Comment by philmmanjaro** [comment]
At **iso-profiles** we just have an additional file. So no branch is needed. For now all is fine as both ways will do the same thing in the end. @udeved just added another layer. It's a blue or red question like in the Matrix. Let him test it. However, I see this earliest to been merged, if ever, within 0.9.14 or later ...

**Comment by udeved** [comment]
Its simply a matter of being able to enable additional layer by dropping a file of symlink in profile.
No file, no layer additionally, but its clean instead of already growing number of package lists in shared again.
Its the same mechanism we use for all other layers too.

**Comment by AJSlye** [comment]
Ok for now I'll just ignore these changes. ;)

**Comment by udeved** [comment]
Whatever, since free drivers are only 10MB increase, I think I will go for preinstalled free drivers and prop only in mhwd repo. I'll see how that goes.

**Comment by AJSlye** [comment]
That's basically what I did on my ISO as well with my Packages-Mhwd-Xorg file:

### General Packages

>openrc manjaro-livecd-openrc
>systemd manjaro-livecd-systemd
#manjaro-livecd-cli-installer
manjaro-tools-base		# recovery chroot tools
calamares			# new GUI installer of manjaro
#calamares-dev			# new GUI installer of manjaro - kpmcore
#thus
#thus-dev
#gparted

### Driver set for fast live-session bootup (optional)

# Xorg Drivers
xf86-video-apm
xf86-video-ark
xf86-video-ati
xf86-video-chips
xf86-video-dummy
xf86-video-fbdev
xf86-video-glint
xf86-video-i128
xf86-video-i740
xf86-video-intel
xf86-video-mach64
xf86-video-neomagic
xf86-video-nouveau
xf86-video-nv
xf86-video-openchrome
xf86-video-r128
xf86-video-rendition
xf86-video-s3
xf86-video-s3virge
xf86-video-savage
xf86-video-siliconmotion
xf86-video-sis
xf86-video-sisusb
xf86-video-tdfx
xf86-video-trident
xf86-video-tseng
xf86-video-vesa
xf86-video-vmware
xf86-video-voodoo

# Bumblebee Support
>systemd bumblebee
>openrc bumblebee-openrc
primus
>multilib lib32-primus
KERNEL-bbswitch

# Network Support
KERNEL-broadcom-wl
KERNEL-rt3562sta
KERNEL-r8168

# Proprietary Drivers (nvidia only)
#>nonfree_default nvidia-utils
#>nonfree_default KERNEL-nvidia
#>nonfree_multilib lib32-nvidia-utils

# Virtualbox Drivers
>openrc virtualbox-guest-utils-openrc
>systemd virtualbox-guest-utils
KERNEL-virtualbox-guest-modules

# Xorg Packages
libxvmc

# Xorg x86_64 Packages
>multilib lib32-libxvmc

# VMWare Support
open-vm-tools
xf86-input-vmmouse

**Comment by udeved** [comment]
I just added instructions for calamares to unpack an optional drivers image

**Comment by AJSlye** [comment]
So now there is two repositories of packages?

**Comment by udeved** [comment]
Nope, I just tell calamares to also unpack the optional drivers image with the free drivers, since the mhwd repo only contains prop. drivers, so that I don't get missing drivers source for mhwd in any case.
Ie the mdhwd drivers image is installed too. Its experimental, haven't done a test build and install yet.

**Comment by AJSlye** [comment]
OK but before all the drivers were in the one layer, you separated them then?
And is so, where are the network, virtual box, vmware and bumblebee packaeges now?

**Comment by udeved** [comment]
I haven't done any package list work yet, I would do it in my profile anyway first.
This optional layer stuff is not finished yet.

**Comment by udeved** [comment]
The idea is to get perhaps around shared/Packages-Desktop in favor of an additional layer with xorg all profiles need.

**Comment by udeved** [comment]
My thinking behind this, unpacking the sqfs file depends on cpu. Most systems handle it sufficiently.
However, if I try to split for example the quite big desktop image in a xorg and a desktop sqfs, it may be faster to unpack. I think the mounting of the layers isn't the issue, and also squashing the images should improve on a weaker cpu.

**Comment by AJSlye** [comment]
Are you suggesting that what your trying to do is make the ISO boot faster using less resources than what we have now? If so than I say excellent.

**Comment by philmmanjaro** [comment]
What ever the outcome here will be, it has to been tested, discussed by the team and agreed that we will go this way. Also it should not conflict with the current status. So please test it thru and thru and create a merge-request later one, so we all can discuss and test it. Remember, if you change your profile and somebody else tries to build it with the old way it might be broken. Maybe a branch for **iso-profiles** is therefor also needed ...

**Comment by udeved** [comment]
I testing profile(actually a copy of my profile) is fully sufficient to juggle with xorg packages in a new list.

**Comment by philmmanjaro** [comment]
Ah, ok. Keep me updated on your findings ;)

**Comment by udeved** [comment]
You think its doable? Ie having only prop. drivers in repo, and free drivers preinstalled as well as installed on target system? atm, half the free drivers are already installed, such as the xorg-input drivers.
I personally only use xorg-input-evdev and always remove keyboard and mouse xorg driver.

**Comment by philmmanjaro** [comment]
Only install the graphical drivers on a live-session overlay, not on images getting unpacked to target. Else we would have installed drivers the user doesn't need at all. Free drivers don't conflict but I don't want to have them pre-installed on the target system. **mhwd** should detect and only install the needed. So **mhwd-repository** still need to hold all drivers, doesn't matter if free or non-free.

**Comment by udeved** [comment]
Either that, or optionally install free drivers on target.
If I create a repo with non free drivers only which depends on mhwd list, I need the free drivers installed so that I don't get error if not online. Right?
Its a trade off to consider, +10MB, but faster boot and non free driver blacklists free driver, at least nvidia does.

**Comment by philmmanjaro** [comment]
We have all needed drivers in the **mhwd-repository** as we use that repo instead the one we have on our servers. So **mhwd** basically does always an offline installation. All drivers are needed to be in that repository. Splitting it is not possible.

**Comment by philmmanjaro** [comment]
It doesn't matter in which overlay you put the pre-installed packages to have a faster boot, but the packages for the **mhwd-repo** should not been changed at all.
